lib/jekyll/site.rb

```diff
@@ -158,7 +164,6 @@ def render
         end
       end
 
-      payload = site_payload
```

Comment:
@parkr: I probably need to clarify the problem a bit more. I removed this line because it overwrites the value of the local variable 'payload' which can be altered by site pre_render hooks. There's a bunch of possible solutions to this problem but in order to suggest one, I think I need to understand this performance issue.